scrape/scrape.go

```diff
@@ -189,7 +189,7 @@ func newScrapePool(cfg *config.ScrapeConfig, app Appendable, logger log.Logger)
 		loops:         map[uint64]loop{},
 		logger:        logger,
 	}
-	sp.newLoop = func(t *Target, s scraper, limit int, honor bool, mrc []*relabel.Config) loop {
+	sp.newLoop = func(t *Target, s scraper, limit int, honorLabels, honorTimestamps bool, mrc []*relabel.Config) loop {
```

Comment:
I think we're at the point where we should have a struct for all these options.

Reply:
Right. I will do it in two steps: one pr with the sctuct only and I will rebase this PR on top of that one.